tiatoolbox/models/classification/patch_predictor.py

```diff
@@ -206,20 +202,107 @@ def __init__(
         """
         super().__init__()
 
-        if model is None and predefined_model is None:
-            raise ValueError("Must provide either of `model` or `predefined_model`")
+        self.img_list = None
+        self.output_list = None
+        self.mode = None
+
+        if model is None and pretrained_model is None:
+            raise ValueError("Must provide either of `model` or `pretrained_model`")
 
         if model is not None:
             self.model = model
+            iostate = None  # retrieve iostate from provided model ?
         else:
-            self.model = get_predefined_model(predefined_model, pretrained_weight)
+            model, iostate = get_pretrained_model(pretrained_model, pretrained_weight)
 
+        self._iostate = iostate  # for storing original
+        self.iostate = None  # for runtime
+        self.model = model  # for runtime, such as after wrapping with nn.DataParallel
+        self.pretrained_model = pretrained_model
         self.batch_size = batch_size
         self.num_loader_worker = num_loader_worker
         self.verbose = verbose
 
     @staticmethod
-    def __postprocess(probabilities):
+    def merge_predictions(
+        img,
+        output,
+        resolution=None,
+        units=None,
+    ):
+        """Merge patch-level predictions to form a 2-dimensional prediction map.
+
+        Args:
+            img (:obj:`str` or :obj:`pathlib.Path` or :class:`numpy.ndarray`):
+                A HWC image or a path to WSI.
+            output_list (list): List of outputs generated by the model.
```

Comment:
it seems that the right argument here is a single `output` not `output_list`. Consider correcting the docstring.